rfcs/testing.md

```diff
@@ -0,0 +1,75 @@
+# RFC: Testing standards
+
+---
+
+_@andrefcdias_
+
+## Summary
+
+The purpose of this RFC is to standardize our testing process and clarify the purpose of each of the testing layers we have. It is supposed to define what is the intended result from tests and why they are required.
+
+## Background
+
+This RFC came from the discussion of unit testing styles in components such as Flex and Text and the debate of having a valuable test vs a "correct" one.
+
+## Problem statement
+
+At the moment, everyone has their own testing standards and there is no common approach to problems like components that only apply styles and need to be tested.
+There is no majority decision on the solution for this problem, being split between Visual Regression tests (VR tests) and Unit Tests (UT).
+
+By defining a common approach, we can make the testing process less ambiguous, as well as provide a guide for the community to follow.
+
+## Detailed Design or Proposal
+
+To start, this is the proposal for defining the goals of each testing layer.
+
+- **Conformance**
+
+  Ensuring the consistency of our components. They test that props like `as`, `ref`, `className` are supported and that the component is rendered.
+
+  These can be achieved simply through the `isConformant` function generated when creating a converged component.
+
+- **DOM Snapshot**
+
+  Stricly DOM structure consistency enforcing. This will guarantee we do not introduce breaking changes for users who expect a certain structure to be rendered by our components.
+
+  They are achieved with Jest's `expect(component).toMatchInlineSnapshot(...)`. This is preferred over `.toMatchSnapshot` as it is immediately explicit what is expected from the test.
```

Comment:
I'm not sure how practical this is for more complex components since giant inline snapshots (including expanded styles) would make the test files really hard to read. If we want to commit to inline snapshots, we'd need to move nontrivial snapshot tests to separate files and/or place strict limits on what should be snapshot-tested.

Reply:
IMO giant snapshots are a testing antipattern, and my experience is that reviewers gloss over any giant snapshots. In those cases you should be asserting instead.

Reply:
I agree, if we have a giant inline snapshot, then something else is wrong. Should that be included in this RFC?

Reply:
this 👇 -> 100%

> IMO giant snapshots are a testing antipattern, and my experience is that reviewers gloss over any giant snapshots. In those cases you should be asserting instead.